### PR TITLE
PHPStan: Enable more checks

### PR DESCRIPTION
**Pull request by Jellyfrog**
Please give a short description what your pull request is for

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [ ] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by Jellyfrog** [comment]
@murrant what do you say about this error ; https://github.com/librenms/librenms/pull/14318/files#file-librenms-device-processor-php-L145

```
        // if no processors found, check OS discovery (which will fall back to HR and UCD if not implemented
        if (empty($processors) && $os instanceof ProcessorDiscovery) {
```
> Instanceof between LibreNMS\OS and LibreNMS\Interfaces\Discovery\ProcessorDiscovery will always evaluate to true.

Since OS itself implements ProcessorDiscovery, it doesn't really get overridden by a custom os?

**Comment by murrant** [comment]
@Jellyfrog all OS inherit the OS class. Some may implement custom processor polling, overriding that.

The error is correct, the if check can be removed.

File: LibreNMS/Proc.php [hunk 1]
```diff
@@ -260,9 +260,6 @@ public function getStatus()
      */
     public function isRunning()
     {
-        if (! is_resource($this->_process)) {
```
Comment (Jellyfrog) [inline LibreNMS/Proc.php:263]: It has this check in __construct which I guess is why phpstan complains:
```
        if (! is_resource($this->_process)) {
            throw new Exception("Command failed: $cmd");
        }
```

File: LibreNMS/Validations/Poller/CheckRedis.php [hunk 2]
```diff
@@ -62,6 +62,7 @@ public function enabled(): bool
     private function redisIsAvailable(): bool
     {
         try {
+            /* @phpstan-ignore-next-line */
             Redis::ping();
```
Comment (Jellyfrog) [inline LibreNMS/Validations/Poller/CheckRedis.php:66]: > Call to an undefined static method Illuminate\Support\Facades\Redis::ping().
These needs to be stubbed in larastan, think its fixed in laravel9

File: app/Models/UserPref.php [hunk 3]
```diff
@@ -99,9 +99,6 @@ protected function setKeysForSaveQuery($query)
     {
         /** @var array */
         $keys = $this->getKeyName();
-        if (! is_array($keys)) {
```
Comment (Jellyfrog) [inline app/Models/UserPref.php:102]: > Call to function is_array() with array will always evaluate to true.

The key is an array, not sure what it's suppose to check?

`protected $primaryKey = ['user_id', 'pref'];`

File: app/Providers/ErrorReportingProvider.php [hunk 4]
```diff
@@ -46,6 +46,7 @@ class ErrorReportingProvider extends \Facade\Ignition\IgnitionServiceProvider
 
     public function boot(): void
     {
+        /* @phpstan-ignore-next-line */
         if (! method_exists(\Facade\FlareClient\Flare::class, 'filterReportsUsing')) {
```
Comment (Jellyfrog) [inline app/Providers/ErrorReportingProvider.php:50]: >  Call to function method_exists() with 'Facade\\FlareClient\\Flare' and 'filterReportsUsing' will always evaluate to true.

Lets just ignore this, its for people who didn't update.

File: LibreNMS/Alert/Transport/Elasticsearch.php [hunk 5]
```diff
@@ -61,7 +61,7 @@ public function contactElasticsearch($obj, $opts)
         }
 
         if (! empty($opts['es_index'])) {
-            $index = strftime($opts['es_index']);
+            $index = date($opts['es_index']);
```
Comment (Jellyfrog) [inline LibreNMS/Alert/Transport/Elasticsearch.php:64]: This change might break elastic transport

Comment (murrant) [reply LibreNMS/Alert/Transport/Elasticsearch.php:64]: Maybe strtotime() is what was supposed to be here?

Comment (Jellyfrog) [reply LibreNMS/Alert/Transport/Elasticsearch.php:64]: No its to build a index name for elasticsearch

Comment (Jellyfrog) [reply LibreNMS/Alert/Transport/Elasticsearch.php:64]: The default string is the change above it

**Review by murrant — APPROVED** [review]

**Pull request merged**